amazon-s3/6.0/modules/ROOT/pages/amazon-s3-connector-reference.adoc

```diff
@@ -0,0 +1,3652 @@
+| Upload ID a| String |  Upload ID identifying the multipart upload whose parts are listed. |  | x
+| Page Size a| Number |  Sets the page size of the response. | `1000` |
+| Part number marker a| Number |  Specifies the part after which listing should begin. Parts only with higher part numbers are listed. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
+| Streaming Strategy a| * <<RepeatableInMemoryIterable>>
+* <<RepeatableFileStoreIterable>>
+* non-repeatable-iterable |  Configures how Mule processes streams. Repeatable streams are the default behavior. |  |
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  `#[payload]` |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+==== Output
+
+[%autowidth.spread]
+|===
+|Type |Array of <<ApiObjectPart>>
+|===
+
+==== Associated Configurations
+
+* <<config>>
+
+
+
+[[PutBucketAcl]]
+=== Put Bucket Acl
+`<s3:put-bucket-acl>`
+
+
+Sets the permissions on an existing bucket using access control lists (ACL). To set the ACL of a bucket, you must have the WRITE_ACP permission. You can use one of the following two ways to set a bucket's permissions:
+
+* Specify the ACL in the request body
+* Specify permissions using request headers
+
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Bucket a| String |  Name of the bucket. |  | x
+| Grants a| Array of <<Grant>> |  List of grants for this bucket. |  |
+| Owner Id a| String |  Canonical AWS ID of the owner. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+
+==== Associated Configurations
+
+* <<config>>
+
+==== Throws
+
+* S3:BAD_REQUEST
+* S3:CONNECTIVITY
+* S3:FORBIDDEN
+* S3:RETRY_EXHAUSTED
+* S3:S3_ERROR
+* S3:SERVER_BUSY
+* S3:SERVICE_ERROR
+
+
+[[PutBucketCors]]
+=== Put Bucket Cors
+`<s3:put-bucket-cors>`
+
+
+Sets the CORS configuration for your bucket. If the configuration exists, Amazon S3 replaces it.
+
+To use this operation, you must be allowed to perform the s3:PutBucketCORS action. By default, the bucket owner has this permission and can grant it to others.
+
+You set this configuration on a bucket so that the bucket can service cross-origin requests. For example, you might want to enable a request whose origin is \http://www.example.com to access your Amazon S3 bucket at \my.example.bucket.com by using the browser's XMLHttpRequest capability.
```

Comment:
The URLS should be in back ticks with plusses (so they don't look like real links in the published output).